### PR TITLE
Initial support for Xilinx Kria

### PR DESCRIPTION
**Pull request by ricardosalveti**
Enough support for making the image compatible with the standard firmware that is already provided over QSPI.

Remaining work:
- Validate our own spl/u-boot
- Identify the flash offset for the bootloader (https://xilinx-wiki.atlassian.net/wiki/spaces/A/pages/1641152513/Kria+K26+SOM)
- Validate OP-TEE
- Support scr-fit boot script
- Support multiboot
- Support boot firmware updates

### CODE REVIEW

**Comment by ricardosalveti** [comment]
```
cat /proc/mtd
dev:    size   erasesize  name
mtd0: 00080000 00001000 "Image Selector"
mtd1: 00080000 00001000 "Image Selector Golden"
mtd2: 00020000 00001000 "Persistent Register"
mtd3: 00020000 00001000 "Persistent Register Backup"
mtd4: 000c0000 00001000 "Open_1"
mtd5: 00d00000 00001000 "Image A (FSBL, PMU, ATF, U-Boot)"
mtd6: 00080000 00001000 "ImgSel Image A Catch"
mtd7: 00d00000 00001000 "Image B (FSBL, PMU, ATF, U-Boot)"
mtd8: 00080000 00001000 "ImgSel Image B Catch"
mtd9: 00100000 00001000 "Open_2"
mtd10: 00200000 00001000 "Recovery Image"
mtd11: 00200000 00001000 "Recovery Image Backup"
mtd12: 00020000 00001000 "U-Boot storage variables"
mtd13: 00020000 00001000 "U-Boot storage variables backup"
mtd14: 00010000 00001000 "SHA256"
mtd15: 01db0000 00001000 "User"
```

**Comment by ricardosalveti** [comment]
image_update tooling available for manipulating the a/b image banks if needed.

**Review by angolini — APPROVED** [review]

**Review by MrCry0 — APPROVED** [review]
LGTM

**Review by detsch — APPROVED** [review]

File: meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/u-boot/u-boot-xlnx/k26/lmp-base.cfg [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+CONFIG_SPL_FIT_PRINT=y
+CONFIG_FIT_SIGNATURE=y
+CONFIG_SPL_FIT_SIGNATURE=y
+CONFIG_SPL_RSA_VERIFY=y
+CONFIG_RSA=y
+CONFIG_ENV_IS_NOWHERE=n
+CONFIG_ENV_IS_IN_FAT=y
+CONFIG_ENV_FAT_INTERFACE="mmc"
+CONFIG_ENV_FAT_DEVICE_AND_PART="0:1"
+CONFIG_ENV_OFFSET=0x1440000
```
Comment (igoropaniuk) [inline meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/u-boot/u-boot-xlnx/k26/lmp-base.cfg:15]: I've chosen this offsets based on custom layout I used for boot image on QSPI on Avnet Board.
These values might overlap with something else (from https://xilinx-wiki.atlassian.net/wiki/spaces/A/pages/1641152513/Kria+K26+SOM#Boot-FW-QSPI-Memory-Map, it's `“Image B” - BOOT.BIN`)


`),

File: meta-lmp-bsp/conf/machine/uz3eg-iocc.conf [hunk 2]
```diff
@@ -3,6 +3,7 @@
 #@DESCRIPTION: Machine support for UltraZed-EG with IOCC customized for LmP
 
 DEFAULTTUNE = "cortexa53"
+SOC_VARIANT = "eg"
 
 require conf/machine/include/ultrazed-lmp.inc
```
Comment (igoropaniuk) [inline meta-lmp-bsp/conf/machine/uz3eg-iocc.conf:8]: ok, found reply for my question from above.

Maybe we can replace it with `require conf/machine/include/zynqmp-lmp.inc` instead?

File: meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/u-boot/u-boot-xlnx/k26/lmp-base.cfg (hunk 1 above)
Comment (igoropaniuk) [reply meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/u-boot/u-boot-xlnx/k26/lmp-base.cfg:15]: I think these two configs for ENV offsets/size can be just dropped for now

File: meta-lmp-bsp/conf/machine/include/ultrazed-lmp.inc [hunk 3]
```diff
@@ -2,68 +2,4 @@
```
[the remark this answers is not in the document]
Comment (ricardosalveti) [reply meta-lmp-bsp/conf/machine/include/ultrazed-lmp.inc:2]: Don't want to delete just yet as we got custom (factory specific) targets including this file, which extends the MACHINEOVERRIDES (we use the uz and uz3eg overrides).

File: meta-lmp-bsp/conf/machine/uz3eg-iocc.conf (hunk 2 above)
Comment (ricardosalveti) [reply meta-lmp-bsp/conf/machine/uz3eg-iocc.conf:8]: Yup, planning on following that up on another pr.

File: meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/u-boot/u-boot-xlnx/k26/lmp-base.cfg (hunk 1 above)
Comment (ricardosalveti) [reply meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/u-boot/u-boot-xlnx/k26/lmp-base.cfg:15]: Indeed, will update this one.

File: meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/u-boot/u-boot-xlnx/k26/lmp-base.cfg [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+CONFIG_ENV_OFFSET_REDUND=0x1480000
+CONFIG_TEE=y
+CONFIG_OPTEE=y
+CONFIG_OPTEE_LIB=y
+# Align max size with ATF (OCRAM - 256KB)
+CONFIG_SPL_SIZE_LIMIT=0x25000
+CONFIG_SYS_SPI_BOOT_IMAGE_OFFS2=0x60000
```
[the remark this answers is not in the document]
Comment (ricardosalveti) [reply meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/u-boot/u-boot-xlnx/k26/lmp-base.cfg:22]: Yeah, at the initial porting I basically just copy & pasted to get a working binary (as it is required by the build), but work is still required to get it tested and aligned with the imgsel logic.

**Review by igoropaniuk — APPROVED** [review]
LGTM

**Pull request merged**